fuel-asm/tests/encoding.rs

```diff
@@ -11,261 +10,188 @@ fn opcode() {
-        Opcode::gtf(r, r, GTFArgs::CreateWitnessesCount),
-        Opcode::gtf(r, r, GTFArgs::CreateSalt),
-        Opcode::gtf(r, r, GTFArgs::CreateStorageSlotAtIndex),
-        Opcode::gtf(r, r, GTFArgs::CreateInputAtIndex),
-        Opcode::gtf(r, r, GTFArgs::CreateOutputAtIndex),
-        Opcode::gtf(r, r, GTFArgs::CreateWitnessAtIndex),
-        Opcode::gtf(r, r, GTFArgs::InputType),
-        Opcode::gtf(r, r, GTFArgs::InputCoinTxId),
-        Opcode::gtf(r, r, GTFArgs::InputCoinOutputIndex),
-        Opcode::gtf(r, r, GTFArgs::InputCoinOwner),
-        Opcode::gtf(r, r, GTFArgs::InputCoinAmount),
-        Opcode::gtf(r, r, GTFArgs::InputCoinAssetId),
-        Opcode::gtf(r, r, GTFArgs::InputCoinTxPointer),
-        Opcode::gtf(r, r, GTFArgs::InputCoinWitnessIndex),
-        Opcode::gtf(r, r, GTFArgs::InputCoinMaturity),
-        Opcode::gtf(r, r, GTFArgs::InputCoinPredicateLength),
-        Opcode::gtf(r, r, GTFArgs::InputCoinPredicateDataLength),
-        Opcode::gtf(r, r, GTFArgs::InputCoinPredicate),
-        Opcode::gtf(r, r, GTFArgs::InputCoinPredicateData),
-        Opcode::gtf(r, r, GTFArgs::InputContractTxId),
-        Opcode::gtf(r, r, GTFArgs::InputContractOutputIndex),
-        Opcode::gtf(r, r, GTFArgs::InputContractBalanceRoot),
-        Opcode::gtf(r, r, GTFArgs::InputContractStateRoot),
-        Opcode::gtf(r, r, GTFArgs::InputContractTxPointer),
-        Opcode::gtf(r, r, GTFArgs::InputContractId),
-        Opcode::gtf(r, r, GTFArgs::InputMessageId),
-        Opcode::gtf(r, r, GTFArgs::InputMessageSender),
-        Opcode::gtf(r, r, GTFArgs::InputMessageRecipient),
-        Opcode::gtf(r, r, GTFArgs::InputMessageAmount),
-        Opcode::gtf(r, r, GTFArgs::InputMessageNonce),
-        Opcode::gtf(r, r, GTFArgs::InputMessageWitnessIndex),
-        Opcode::gtf(r, r, GTFArgs::InputMessageDataLength),
-        Opcode::gtf(r, r, GTFArgs::InputMessagePredicateLength),
-        Opcode::gtf(r, r, GTFArgs::InputMessagePredicateDataLength),
-        Opcode::gtf(r, r, GTFArgs::InputMessageData),
-        Opcode::gtf(r, r, GTFArgs::InputMessagePredicate),
-        Opcode::gtf(r, r, GTFArgs::InputMessagePredicateData),
-        Opcode::gtf(r, r, GTFArgs::OutputType),
-        Opcode::gtf(r, r, GTFArgs::OutputCoinTo),
-        Opcode::gtf(r, r, GTFArgs::OutputCoinAmount),
-        Opcode::gtf(r, r, GTFArgs::OutputCoinAssetId),
-        Opcode::gtf(r, r, GTFArgs::OutputContractInputIndex),
-        Opcode::gtf(r, r, GTFArgs::OutputContractBalanceRoot),
-        Opcode::gtf(r, r, GTFArgs::OutputContractStateRoot),
-        Opcode::gtf(r, r, GTFArgs::OutputMessageRecipient),
-        Opcode::gtf(r, r, GTFArgs::OutputMessageAmount),
-        Opcode::gtf(r, r, GTFArgs::OutputContractCreatedContractId),
-        Opcode::gtf(r, r, GTFArgs::OutputContractCreatedStateRoot),
-        Opcode::gtf(r, r, GTFArgs::WitnessDataLength),
-        Opcode::gtf(r, r, GTFArgs::WitnessData),
-        Opcode::Undefined,
+    let mut instructions = vec![
```

Comment:
would there be a way to autogenerate this based on the macro to ensure no cases are missed?

Reply:
Implemented here https://github.com/FuelLabs/fuel-vm/pull/340, a separate review for that would be nice